app/Util/DocuSign.php

```diff
@@ -161,4 +173,232 @@ private static function getAccessToken(): string
+                    ]
+                )
+                ->setIncludeEnvelopeVoidReason(false)
+                ->setLoggingEnabled(true)
+                ->setRequireAcknowledgment(true)
+                ->setUrl(
+                    URL::signedRoute('webhook-client-docusign', ['internalEnvelopeId' => $envelope->id])
+                ),
+        ];
+    }
+
+    /**
+     * Build an EnvelopeDefinition for a membership agreement where only the member needs to sign.
+     *
+     * @phan-suppress PhanTypeMismatchArgumentProbablyReal
+     */
+    private static function membershipAgreementEnvelopeDefinitionForMemberOnly(
+        DocuSignEnvelope $envelope
+    ): EnvelopeDefinition {
+        return (new EnvelopeDefinition())
+            ->setStatus('sent')
+            ->setTemplateId(config('docusign.templates.membership_agreement_member_only'))
+            ->setTemplateRoles(
+                [
+                    self::membershipAgreementTemplateRoleForMember($envelope->signedBy),
+                ]
+            )
+            ->setEmailSubject('RoboJackets Membership Agreement for '.$envelope->signedBy->full_name)
+            ->setEmailBlurb(trim(view('mail.agreement.notificationformember')->render()))
+            ->setEmailSettings(
+                (new EmailSettings())
+                    ->setReplyEmailAddressOverride('support@robojackets.org')
+                    ->setReplyEmailNameOverride('RoboJackets')
+            )->setNotification(
+                (new Notification())
+                    ->setUseAccountDefaults(false)
+                    ->setReminders(
+                        (new Reminders())
+                            ->setReminderEnabled(true)
+                            ->setReminderDelay(2)
+                            ->setReminderFrequency(2)
+                    )
+                    ->setExpirations(
+                        (new Expirations())
+                            ->setExpireEnabled(true)
+                            ->setExpireWarn(10)
+                            ->setExpireAfter(60)
+                    )
+            )
+            ->setAllowComments(false)
+            ->setAllowMarkup(false)
+            ->setAllowReassign(false)
+            ->setAllowRecipientRecursion(false)
+            ->setAllowViewHistory(true)
+            ->setAutoNavigation(false)
+            ->setEnableWetSign(true)
+            ->setEnvelopeIdStamping(true)
+            ->setEventNotifications(self::eventNotifications($envelope))
+            ->setUseDisclosure(true);
+    }
+
+    /**
+     * Build an EnvelopeDefinition for a membership agreement where the member and their parent or guardian both need
+     * to sign.
+     *
+     * @phan-suppress PhanTypeMismatchArgumentProbablyReal
+     */
+    private static function membershipAgreementEnvelopeDefinitionForMemberAndParentOrGuardian(
+        DocuSignEnvelope $envelope
+    ): EnvelopeDefinition {
+        return (new EnvelopeDefinition())
```

Comment:
This can use the other envelope definition as a starting point instead of a new blank one